tests/admin_changelist/tests.py

```diff
@@ -1540,6 +1541,44 @@ def test_search_help_text(self):
             'autofocus aria-describedby="searchbar_helptext">',
         )
 
+    def test_changelist_view_atomicity(self):
+        a = Swallow.objects.create(origin="Swallow A", load=4, speed=1)
+        b = Swallow.objects.create(origin="Swallow B", load=2, speed=2)
+
+        changelist_url = reverse("admin:admin_changelist_swallow_changelist")
+        m = SwallowAdmin(Swallow, custom_site)
+        data = {
+            "form-TOTAL_FORMS": "2",
+            "form-INITIAL_FORMS": "2",
+            "form-MIN_NUM_FORMS": "0",
+            "form-MAX_NUM_FORMS": "1000",
+            "form-0-uuid": str(a.pk),
+            "form-1-uuid": str(b.pk),
+            "form-0-load": "9.0",
+            "form-0-speed": "9.0",
+            "form-1-load": "5.0",
+            "form-1-speed": "5.0",
+            "_save": "Save",
+        }
+
+        request = self.factory.post(changelist_url, data=data)
+        request.user = self.superuser
+
+        with mock.patch.object(
+            m, "save_related", side_effect=DatabaseError
+        ), mock.patch.object(m, "log_change", side_effect=DatabaseError):
```

Comment:
Do we need to mock `log_change()`? Should not be reachable when raising a database error in `save_related()`.

Reply:
So, should I only mock `log_change()`, as it would execute the last database query in the transaction?

Reply:
> So, should I only mock `log_change()`, as it would execute the last database query in the transaction?

Makes sense to me.

Reply:
Okay, will update the testcase.